x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/AbstractRemoteClusterSecurityWithMultipleRemotesRestIT.java

```diff
@@ -0,0 +1,225 @@
+            indexDocRequest.setJsonEntity("{\"name\": \"doc1\"}");
+            assertOK(client().performRequest(indexDocRequest));
+
+            // Search across local cluster and both remotes
+            searchAndAssertIndicesFound(
+                String.format(
+                    Locale.ROOT,
+                    "/local_index,%s:%s/_search?ccs_minimize_roundtrips=%s",
+                    randomFrom("my_remote_*", "*"),
+                    randomFrom("*_index1", "*"),
+                    randomBoolean()
+                ),
+                "cluster1_index1",
+                "cluster2_index1",
+                "local_index"
+            );
+
+            // Search across both remotes using cluster alias wildcard
+            searchAndAssertIndicesFound(
+                String.format(
+                    Locale.ROOT,
+                    "/%s:%s/_search?ccs_minimize_roundtrips=%s",
+                    randomFrom("my_remote_*", "*"),
+                    randomFrom("*_index1", "*"),
+                    randomBoolean()
+                ),
+                "cluster1_index1",
+                "cluster2_index1"
+            );
+
+            // Search across both remotes using explicit cluster aliases
+            searchAndAssertIndicesFound(
+                String.format(
+                    Locale.ROOT,
+                    "/my_remote_cluster:%s,my_remote_cluster_2:%s/_search?ccs_minimize_roundtrips=%s",
+                    randomFrom("cluster1_index1", "*_index1", "*"),
+                    randomFrom("cluster2_index1", "*_index1", "*"),
+                    randomBoolean()
+                ),
+                "cluster1_index1",
+                "cluster2_index1"
+            );
+
+            // Search single remote
+            final boolean searchFirstCluster = randomBoolean();
+            final String index1 = searchFirstCluster ? "cluster1_index1" : "cluster2_index1";
+            searchAndAssertIndicesFound(
+                String.format(
+                    Locale.ROOT,
+                    "/%s:%s/_search?ccs_minimize_roundtrips=%s",
+                    searchFirstCluster ? "my_remote_cluster" : "my_remote_cluster_2",
+                    randomFrom(index1, "*_index1", "*"),
+                    randomBoolean()
+                ),
+                index1
+            );
+
+            // To simplify the test setup, we only ever (randomly) set skip_unavailable on the other remote, not on both
+            // This impacts below failure scenarios; in some cases, skipping the other remote results in overall request success
+            final boolean skipUnavailableOnOtherCluster = isSkipUnavailable("my_remote_cluster_2");
+
+            // Search when one cluster throws 403
+            // No permissions for this index name, so searching for it on either remote will result in 403
+            final String missingIndex = "index1";
```

Comment:
Nit: I found this helps reading the code
```suggestion
            final String missingIndex = "missingIndex";
```